File: chess-website/src/models/chessEngine/gameStateResource.ts
import { attackedSquare } from "./attackedSquare";
import { snapshot } from "./snapshot";
import { square } from "./square";
import { stateInfo } from "./stateInfo";

export interface gameStateResource extends snapshot {
    attacks: attackedSquare[];
    history: snapshot[];
    squares: square[];
    stateInfo: stateInfo;
    fen: string;
} 

File: chess-webapi/Models/GameStateResource.cs
using chess_engine.Models;
using System.Collections.Generic;

namespace chess_webapi.Models
{
	public class GameStateResource : Snapshot
	{
		public List<AttackedSquare> Attacks { get; set; } = new List<AttackedSquare>();
		public List<Snapshot> History { get; set; } = new List<Snapshot>();
		public List<Square> Squares { get; set; } = new List<Square>();
		public StateInfo StateInfo { get; set; }
		public string FEN { get; set; }
	}
}